dokka-subprojects/analysis-kotlin-descriptors-compiler/src/main/kotlin/org/jetbrains/kotlin/serialization/deserialization/AnnotationDeserializer.kt

```diff
@@ -0,0 +1,133 @@
+
+        return when (value.type) {
+            Type.BYTE -> value.intValue.toByte().letIf(isUnsigned, ::UByteValue, ::ByteValue)
+            Type.CHAR -> CharValue(value.intValue.toInt().toChar())
+            Type.SHORT -> value.intValue.toShort().letIf(isUnsigned, ::UShortValue, ::ShortValue)
+            Type.INT -> value.intValue.toInt().letIf(isUnsigned, ::UIntValue, ::IntValue)
+            Type.LONG -> value.intValue.letIf(isUnsigned, ::ULongValue, ::LongValue)
+            Type.FLOAT -> FloatValue(value.floatValue)
+            Type.DOUBLE -> DoubleValue(value.doubleValue)
+            Type.BOOLEAN -> BooleanValue(value.intValue != 0L)
+            Type.STRING -> StringValue(nameResolver.getString(value.stringValue))
+            Type.CLASS -> KClassValue(nameResolver.getClassId(value.classId), value.arrayDimensionCount)
+            Type.ENUM -> EnumValue(nameResolver.getClassId(value.classId), nameResolver.getName(value.enumValueId))
+            Type.ANNOTATION -> AnnotationValue(deserializeAnnotation(value.annotation, nameResolver))
+            Type.ARRAY -> ConstantValueFactory.createArrayValue(
+                value.arrayElementList.map { resolveValue(builtIns.anyType, it, nameResolver) },
+                expectedType
+            )
+            else -> error("Unsupported annotation argument type: ${value.type} (expected $expectedType)")
+        }
+    }
+
+    // This method returns false if the actual value loaded from an annotation argument does not conform to the expected type of the
+    // corresponding parameter in the annotation class. This usually means that the annotation class has been changed incompatibly
+    // without recompiling clients, in which case we prefer not to load the annotation argument value at all, to avoid constructing
+    // an incorrect model and breaking some assumptions in the compiler.
+    private fun doesValueConformToExpectedType(result: ConstantValue<*>, expectedType: KotlinType, value: Value): Boolean {
+        return when (value.type) {
+            Type.CLASS -> {
+                val expectedClass = expectedType.constructor.declarationDescriptor as? ClassDescriptor
+                // We could also check that the class value's type is a subtype of the expected type, but loading the definition of the
+                // referenced class here is undesirable and may even be incorrect (because the module might be different at the
+                // destination where these constant values are read). This can lead to slightly incorrect model in some edge cases.
+                expectedClass == null || KotlinBuiltIns.isKClass(expectedClass)
+            }
+            Type.ARRAY -> {
+                check(result is ArrayValue && result.value.size == value.arrayElementList.size) {
+                    "Deserialized ArrayValue should have the same number of elements as the original array value: $result"
+                }
+
+                // TODO: PATCH START
```

Comment:
this is the only change to original file

Reply:
Can you elaborate on what the patch is in KDoc? Where was it copied from?

Reply:
I've added the link to the changes landed into kotlin.git and rephrased KDoc a bit